flow/shared/authenticator.go

```diff
@@ -0,0 +1,215 @@
+package shared
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type AuthenticationConfig struct {
+	KeySetJSON    string `json:"key_set_json" yaml:"key_set_json" mapstructure:"key_set_json"`
+	Auth0Domain   string `json:"auth0_domain" yaml:"auth0_domain" mapstructure:"auth0_domain"`
+	Auth0ClientID string `json:"auth0_client_id" yaml:"auth0_client_id" mapstructure:"auth0_client_id"`
+	Enabled       bool   `json:"enabled" yaml:"enabled" mapstructure:"enabled"`
+}
+
+type identityProvider struct {
+	keySet      jwk.Set
+	validateOpt jwt.ValidateOption
+	issuer      string
+}
+
+func AuthGrpcMiddleware() ([]grpc.ServerOption, error) {
+	keySetJSON := os.Getenv("PEERDB_OAUTH_KEYSET_JSON")
+	auth0Domain := os.Getenv("PEERDB_OAUTH_AUTH0DOMAIN")
+	auth0ClientID := os.Getenv("PEERDB_OAUTH_AUTH0CLIENTID")
```

Comment:
can we move this to `peerdbenv/env.go` please? Something like `PeerDBGetOAuthConfig` which returns a struct?